Application/src/visuals/Message.java

```diff
@@ -95,6 +94,39 @@ else if(keepAnimating && node1.getX() > node2.getX()){ // Sending a return messa
                if((animationBounds -= (class_size/6 +curve)) < (this.node2.getX())  - this.node1.getX())
                    keepAnimating = false;
            }
+        //Checks if we are supposed to keep animating, set animationBounds according to how diagramClasses are scaled.
+        else if (keepAnimating && node1.getX() == node2.getX()) { //Sending a self referencing message
+            switch (selfCallCounter) {
+                case 1:
+                    switchDirection = true; // Switch state of the dragon's flying direction
+                    // Move the message to the right of the class
+                    if ((animationBounds += (class_size / 6)) > this.class_size) {
+                        switchDirection = false; // Original state of the dragon's flying direction
+                        selfCallCounter = 2;
+                    }
+                    break;
+
+                case 2:
+                    // Multiple increments to lower the message vertically
+                    for (int i = 0; i < 3; i++) {
+
+                        offset += 8;
+                    }
+                    selfCallCounter = 3;
+                    break;
+
+                case 3:
+                    // move back to starting point
+                    if ((animationBounds -= (class_size / 6)) < 1) {
+                        keepAnimating = false;
+                    }
+                    break;
+
+                //default case
+                default :
+                    selfCallCounter =1;
+            }
+        }
```

Comment:
there's got to be a better way to change selfCallCounter; I mean, it works but typically, switch statements aren't used like this; is there any particular reason why you chose to implement it like that?

Reply:
@gusgolko i used a switch case to separate the different cases i have which are dependent on each other's sequence. your concern is noted and added to the defect backlog for further improvement.